src/common.h

```diff
@@ -184,4 +188,47 @@ Join(const T& container, const std::string& delim)
   return ss.str();
 }
 
+
+// Used by Python Bindings to accept arguments to initialize Frontends.
+using VariantType = std::variant<int, bool, std::string>;
+using UnorderedMapType = std::unordered_map<std::string, VariantType>;
+
+// void
+// printVariant(const VariantType& v)
+// {
+//   std::visit(
+//       [](auto&& arg) {
+//         using T = std::decay_t<decltype(arg)>;
+//         if constexpr (std::is_same_v<T, std::string>) {
+//           std::cout << "Value (string): " << arg << std::endl;
+//         } else if constexpr (std::is_same_v<T, int>) {
+//           std::cout << "Value (int): " << arg << std::endl;
+//         } else if constexpr (std::is_same_v<T, bool>) {
+//           std::cout << "Value (bool): " << std::boolalpha << arg <<
+//           std::endl;
+//         }
+//       },
+//       v);
+// };
+
+template <typename T>
+T
+get_value(const UnorderedMapType& options, const std::string& key)
+{
+  auto curr = options.find(key);
+  bool is_present = (curr != options.end());
+
+  if (!is_present) {
+    throw std::runtime_error("Key: " + key + " not found in options provided.");
```

Comment:
Why are we throwing instead of returning a TRITONSERVER_ERROR?

Reply:
Thank you for the suggestion! I have re-factored the code to have the following: `get_value(...)` returns TRITONSERVER_ERROR*. An Invalid Argument error is given with a custom error message selected depending on the error type.